src/JsonSchema/Constraints/FormatConstraint.php

```diff
@@ -45,74 +45,74 @@ public function check($element, $schema = null, $path = null, $i = null)
                 break;
 
             case 'color':
                 if (!$this->validateColor($element)) {
-                    $this->addError($path, "Invalid color");
+                    $this->addError($path, "Invalid color", 'format', array('format' => $schema->format,));
                 }
                 break;
 
             case 'style':
                 if (!$this->validateStyle($element)) {
-                    $this->addError($path, "Invalid style");
+                    $this->addError($path, "Invalid style", 'format', array('format' => $schema->format,));
                 }
                 break;
 
             case 'phone':
                 if (!$this->validatePhone($element)) {
-                    $this->addError($path, "Invalid phone number");
+                    $this->addError($path, "Invalid phone number", 'format', array('format' => $schema->format,));
                 }
                 break;
 
             case 'uri':
                 if (null === filter_var($element, FILTER_VALIDATE_URL, FILTER_NULL_ON_FAILURE)) {
-                    $this->addError($path, "Invalid URL format");
+                    $this->addError($path, "Invalid URL format", 'format', array('format' => $schema->format,));
                 }
                 break;
 
             case 'email':
                 if (null === filter_var($element, FILTER_VALIDATE_EMAIL, FILTER_NULL_ON_FAILURE)) {
-                    $this->addError($path, "Invalid email");
+                    $this->addError($path, "Invalid email", 'format', array('format' => $schema->format,));
                 }
                 break;
 
             case 'ip-address':
             case 'ipv4':
                 if (null === filter_var($element, FILTER_VALIDATE_IP, FILTER_NULL_ON_FAILURE | FILTER_FLAG_IPV4)) {
-                    $this->addError($path, "Invalid IP address");
+                    $this->addError($path, "Invalid IP address", 'format', array('format' => $schema->format,));
                 }
                 break;
 
             case 'ipv6':
                 if (null === filter_var($element, FILTER_VALIDATE_IP, FILTER_NULL_ON_FAILURE | FILTER_FLAG_IPV6)) {
-                    $this->addError($path, "Invalid IP address");
+                    $this->addError($path, "Invalid IP address", 'format', array('format' => $schema->format,));
                 }
                 break;
 
             case 'host-name':
             case 'hostname':
                 if (!$this->validateHostname($element)) {
-                    $this->addError($path, "Invalid hostname");
+                    $this->addError($path, "Invalid hostname", 'format', array('format' => $schema->format,));
                 }
                 break;
 
             default:
-                $this->addError($path, "Unknown format: " . json_encode($schema->format));
+                $this->addError($path, "Unknown format: " . json_encode($schema->format), 'format', array('format' => $schema->format,));
```

Comment:
Just for clarify. This is not a bug in your PR. The bug already exists in the upstream code, the reason of modify the line is to prevent (supersede) a git conflict with #125